2016-01-02  Helge Deller  <deller@gmx.de>

	[BZ #19285]
	* sysdeps/unix/sysv/linux/hppa/bits/mman.h (MAP_STACK): Define.
	(MAP_HUGETLB, MCL_ONFAULT): Likewise.
	(MADV_HUGEPAGE, MADV_NOHUGEPAGE, MADV_DONTDUMP, MADV_DODUMP): Likewise.
	(MADV_xxK_PAGES): Remove.

--- a/sysdeps/unix/sysv/linux/hppa/bits/mman.h
+++ b/sysdeps/unix/sysv/linux/hppa/bits/mman.h
@@ -58,6 +58,8 @@
 # define MAP_GROWSDOWN	0x8000		/* Stack-like segment */
 # define MAP_POPULATE	0x10000		/* Populate (prefault) pagetables */
 # define MAP_NONBLOCK	0x20000		/* Do not block on IO */
+# define MAP_STACK	0x40000		/* Create for process/thread stacks */
+# define MAP_HUGETLB	0x80000		/* Create a huge page mapping */
 #endif
 
 /* Flags to "msync"  */
@@ -68,6 +70,7 @@
 /* Flags to "mlockall"  */
 #define MCL_CURRENT	1		/* Lock all current mappings */
 #define MCL_FUTURE	2		/* Lock all future mappings */
+#define MCL_ONFAULT	4		/* Lock all pages that are faulted in */
 
 /* Flags for `mremap'.  */
 #ifdef __USE_GNU
@@ -90,18 +93,13 @@
 # define MADV_DOFORK	 11	/* Do inherit across fork.  */
 # define MADV_MERGEABLE   65	/* KSM may merge identical pages */
 # define MADV_UNMERGEABLE 66	/* KSM may not merge identical pages */
+# define MADV_HUGEPAGE	 67	/* Worth backing with hugepages */
+# define MADV_NOHUGEPAGE 68	/* Not worth backing with hugepages */
+# define MADV_DONTDUMP	 69	/* Explicity exclude from the core dump,
+				   overrides the coredump filter bits */
+# define MADV_DODUMP	 70	/* Clear the MADV_NODUMP flag */
 #endif
 
-/* The range 12-64 is reserved for page size specification. */
-#define MADV_4K_PAGES   12              /* Use 4K pages  */
-#define MADV_16K_PAGES  14              /* Use 16K pages */
-#define MADV_64K_PAGES  16              /* Use 64K pages */
-#define MADV_256K_PAGES 18              /* Use 256K pages */
-#define MADV_1M_PAGES   20              /* Use 1 Megabyte pages */
-#define MADV_4M_PAGES   22              /* Use 4 Megabyte pages */
-#define MADV_16M_PAGES  24              /* Use 16 Megabyte pages */
-#define MADV_64M_PAGES  26              /* Use 64 Megabyte pages */
-
 /* The POSIX people had to invent similar names for the same things.  */
 #ifdef __USE_XOPEN2K
 # define POSIX_MADV_NORMAL	0 /* No further special treatment.  */
